### DISCUSSION TITLE
关于websocket 流式 TTS 服务 返回的音频base64

### DISCUSSION THREAD

**Discussion by syy-love** [PaddlePaddle/PaddleSpeech#2073] [Q&A] [answered]
请问返回的base64转成音频文件不成功~哪位大佬给看看
代码：
import base64
import io
import os
import soundfile
from paddlespeech.cli.log import logger
from paddlespeech.server.utils.audio_process import wav2pcm
from paddlespeech.server.utils.util import wav2base64

def postprocess(wav_base64: str, outfile: str) -> float:
        audio_data_byte = base64.b64decode(wav_base64)
        # from byte
        samples, sample_rate = soundfile.read(
            io.BytesIO(audio_data_byte), dtype='float32')

        # transform audio
        if outfile.endswith(".wav"):
            soundfile.write(outfile, samples, sample_rate)
        elif outfile.endswith(".pcm"):
            temp_wav = str(random.getrandbits(128)) + ".wav"
            soundfile.write(temp_wav, samples, sample_rate)
            wav2pcm(temp_wav, outfile, data_type=np.int16)
            os.remove(temp_wav)
        else:
            logger.error("The format for saving audio only supports wav or pcm")

报错：
![HKYG_VV`VLBNPY4GGQMA 25](https://user-images.githubusercontent.com/51405724/175265029-931232b9-e4c8-4f6f-877f-258e88698106.png)

**Comment by lym0302** [comment, marked as answer]
Soundfile first tries to load your system library. So put your compiled libsndfile somewhere on PATH, or in LD_LIBRARY_PATH, or in your current directory. Only when it can't find a system library will soundfile try its own.
来源：https://github.com/bastibe/python-soundfile/issues/302
